src/tainting/Taint.ml

```diff
@@ -193,10 +184,9 @@ type source = {
       *)
   precondition : (taint list * R.precondition) option;
 }
-[@@deriving show]
 
-and orig = Src of source | Var of lval | Control [@@deriving show]
-and taint = { orig : orig; tokens : tainted_tokens } [@@deriving show]
+and orig = Src of source | Var of lval | Control
```

Comment:
why removing those deriving show?
It's pretty convenient for debugging.

Reply:
Removed them because they were dead I think, we are defining and exporting many of those `show_` functions. But yeah maybe I should keep the `deriving show` and rename my to `pretty` or something.

Reply:
you need to add them also in the .mli, otherwise ocamlc will complain indeed that they are dead.
But they are still useful if one wants to add a Logs.debug (fun m -> m "..." (show_taint xxx))
just to debug one thing.